### ISSUE TITLE
Damaged Battleship Stats Bug

### ISSUE THREAD

**Issue by demuetz** [djensen47/AACalc#1]
Wow, the first issue!

Test case: 1 Battleship attacks 1 Destroyer
Expected Outcome: Attacker results contain 3 items (one each for No Losses, Battleship Damaged, Battleship Lost)
Obtained Outcome: 2 items only, one each for No Losses and Battlesip Lost

This is a minor thing I guess (and maybe a known issue), but still... ;-)

**Comment by gmassaro** [comment]
Thanks for the comment! The current %s include damaged battleships and undamaged battleships. The design you observe was purposeful. At the end of combat, per the rulebooks, 2 hit battleships (Bat) that have only 1 hit "heal" fully to 2 hits left. Thus, the program lumps 1hit and 2hit Bats together in the stat line. However, if you run the battle once, then a damaged battleship (dBat) can be listed as a surviving unit. For multiple-run battles, the %s reported are still correct, however. Currently, a battle may indicate 88% of the time a Bat survives. This is, in reality, a combination of dBat and Bat surviving.

If it is desired to see the exact % that a Bat survives the battle with 1 hit, then that should be easy to code in. Again, even though the current results do not indicate a dBat possibility, the %s incorporate that condition (simply treats dBat and Bat the same for adding up the %). Furthermore, keep in mind adding dBat as a possible outcome will increase the number of result lines. In some cases, almost doubling the result bars for multiple runs of the same battle.

Note that one can use a dBat unit as an input to determine more specific %s. Not sure what the priority on this would be given djensen is looking at reworking the GUI for the site. I just wanted to stress that the %s will remain unchanged from the current version of AACalc if we start reporting a damaged Battleship in the outcomes.

**Comment by demuetz** [comment]
Hey gmassaro,

Yeah, I assumed that the 2 percentages are thrown together. I observed this in my own little simulator, with which I'm currently toying around. (I injured my foot and haven't really been able to walk for more than 2 weeks now...) I'm absolutely hearing you about not wanting to make the results unnecessarily long and complex.

I now see that my report was incomplete, sorry. I assumed that the respective rules for 1942 were the same as for 1940. I believe that per these new rules a damaged Battleship can be repaired only at the beginning of the following turn (and only next to a naval base). Thus you want to know, whether a battle leaves you surviving with damaged ships, because obviously your fleet is much more vulnerable. Sorry if I was wrong about the 1942 rules.

By the way: nice work!
